src/scripts/printTask.js

```diff
@@ -0,0 +1,74 @@
+import { getLocal, updateLocal } from './localStorage.js';
+import { getElement, createElement } from './queries.js';
+import newIndex from './updateIndex.js';
+
+function deleteTask(task, tasks) {
+  const objIndex = tasks.findIndex((obj) => obj.index === task.index);
+  const local = getLocal();
+  if (local) {
+    tasks = local;
+  }
+  tasks.splice(objIndex, 1);
+  newIndex(tasks);
+}
+
+function editTask(menu, description, tasks, objIndex) {
+  const local = getLocal();
+  if (local) {
+    tasks = local;
+  }
+  if (menu.innerHTML === '⋮') {
+    menu.innerHTML = '&#10004;';
+    description.contentEditable = true;
+  } else {
+    menu.innerHTML = '⋮';
+    description.contentEditable = false;
+    tasks[objIndex].description = description.innerText;
+  }
+  updateLocal(tasks);
+}
+
+function printTask(task, tasks) {
+  const objIndex = tasks.findIndex((obj) => obj.index === task.index);
+  const li = createElement('li');
+  const done = createElement('input');
+  done.type = 'checkbox';
+  done.checked = task.completed;
+  if (task.completed) li.classList.add('completed');
+  done.addEventListener('change', () => {
+    li.classList.toggle('completed');
+    const local = getLocal();
+    if (local) {
+      tasks = local;
+    }
+    if (done.checked) {
+      tasks[objIndex].completed = true;
+      updateLocal(tasks);
+    } else {
+      tasks[objIndex].completed = false;
+      updateLocal(tasks);
+    }
+  });
+  const description = createElement('span');
+  description.innerHTML = task.description;
+  const menu = createElement('span');
+  menu.innerHTML = '⋮';
+  menu.className = 'task-options';
+  menu.addEventListener('click', () => {
+    editTask(menu, description, tasks, objIndex);
+  });
+  const delBtn = createElement('span');
+  delBtn.innerHTML = '&#10005;';
+  delBtn.className = 'del-btn';
+  delBtn.addEventListener('click', () => {
+    deleteTask(task, tasks);
+    delBtn.parentElement.remove(delBtn);
+  });
```

Comment:
- [ ] You implemented the delete function to update the index, as requested 👏🏽. However, after deleting a random task from the list, the `task.completed` functionality is lost in the updated task. I recorded a [Loom video here](https://www.loom.com/share/4ec44b205edb4b749b5fe576f70379d8) to provide you with some context and how you might resolve it.
Kindly refactor your delete function so that you can still `Clear all completed` tasks after deleting random tasks.